docs/get-started/getting-started-kubernetes.md

```diff
@@ -0,0 +1,256 @@
+     --master $K8SMASTER \
+     --deploy-mode cluster  \
+     --name examplejob \
+     --class org.apache.spark.examples.SparkPi \
+     --conf spark.rapids.sql.concurrentGpuTasks=1 \
+     --conf spark.executor.instances=1 \
+     --conf spark.executor.resource.gpu.amount=1 \
+     --conf spark.executor.memory=4G \
+     --conf spark.executor.cores=1 \
+     --conf spark.task.cpus=1 \
+     --conf spark.task.resource.gpu.amount=1 \
+     --conf spark.rapids.memory.pinnedPool.size=2G \
+     --conf spark.locality.wait=0s \
+     --conf spark.sql.files.maxPartitionBytes=512m \
+     --conf spark.sql.shuffle.partitions=10 \
+     --conf spark.plugins=com.nvidia.spark.SQLPlugin \
+     --conf spark.kubernetes.namespace=$SPARK_NAMESPACE  \
+     --conf spark.kubernetes.driver.pod.name=$SPARK_DRIVER_NAME  \
+     --conf spark.executor.resource.gpu.discoveryScript=/opt/sparkRapidsPlugin/getGpusResources.sh \
+     --conf spark.executor.resource.gpu.vendor=nvidia.com \
+     --conf spark.kubernetes.container.image=$IMAGE_NAME \
+     --conf spark.executor.extraClassPath=/opt/sparkRapidsPlugin/rapids-4-spark_<version>.jar:/opt/sparkRapidsPlugin/cudf-<version>.jar   \
+     --conf spark.driver.extraClassPath=/opt/sparkRapidsPlugin/rapids-4-spark_<version>.jar:/opt/sparkRapidsPlugin/cudf-<version>.jar   \
+     --driver-memory 2G \
+     local:///opt/spark/examples/jars/spark-examples_2.12-3.0.2.jar
+```
+
+   Note: `local://` means the jar file location is inside the Docker image.
+   Since this is `cluster` mode, the Driver is running inside a POD in Kubernetes.
+   When the job is running, we could see both Driver POD and Executor POD(s):
+```shell
+$ kubectl get pods
+NAME                               READY   STATUS    RESTARTS   AGE
+spark-pi-d11075782f399fd7-exec-1   1/1     Running   0          9s
+exampledriver                      1/1     Running   0          15s
+```
+
+   To view the Driver log, use below command:
+```shell
+kubectl logs $SPARK_DRIVER_NAME
+```
+
+   To view the Spark Driver UI when the job is running:
+```shell
+kubectl port-forward $SPARK_DRIVER_NAME 4040:4040
+```
+   Then open web browser:
+```shell
+http://localhost:4040
+```
+
+   To kill the Spark job and delete Driver POD:
+```shell
+$SPARK_HOME/bin/spark-submit --kill spark:$SPARK_DRIVER_NAME
+kubectl delete pod $SPARK_DRIVER_NAME
+```
+
+### spark-shell in `client` mode with an interactive shell
+
+If you need an interactive spark-shell with Executor POD(s) running inside Kubernetes:
+```shell
+$SPARK_HOME/bin/spark-shell \
+     --master $K8SMASTER \
+     --name mysparkshell \
+     --deploy-mode client  \
+     --conf spark.rapids.sql.concurrentGpuTasks=1 \
+     --conf spark.executor.instances=1 \
+     --conf spark.executor.resource.gpu.amount=1 \
+     --conf spark.executor.memory=4G \
+     --conf spark.executor.cores=1 \
+     --conf spark.task.cpus=1 \
+     --conf spark.task.resource.gpu.amount=1 \
+     --conf spark.rapids.memory.pinnedPool.size=2G \
```

Comment:
If we're going to ask for 2GB of pinned memory outside of the executor heap then I think we need to set `spark.kubernetes.memoryOverheadFactor` accordingly so Kubernetes expects the extra memory usage.

Reply:
Added spark.executor.memoryOverhead=3G in each spark-submit.